packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -163,20 +181,32 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
       _state = _DragState.accepted;
       final Offset delta = _pendingDragOffset;
       final Duration timestamp = _lastPendingEventTimestamp;
+      Offset startPosition;
+      Offset updateDelta;
+      switch (dragStartBehavior) {
+        case DragStartBehavior.start:
+          startPosition = _initialPosition + delta;
+          updateDelta = Offset.zero;
+          break;
+        case DragStartBehavior.down:
+          startPosition = _initialPosition;
+          updateDelta = _getDeltaForDetails(delta);
+          break;
+      }
       _pendingDragOffset = Offset.zero;
       _lastPendingEventTimestamp = null;
       if (onStart != null) {
         invokeCallback<void>('onStart', () => onStart(DragStartDetails(
           sourceTimeStamp: timestamp,
-          globalPosition: _initialPosition,
+          globalPosition: startPosition,
         )));
       }
-      if (delta != Offset.zero && onUpdate != null) {
+      if (updateDelta != Offset.zero && onUpdate != null) {
         invokeCallback<void>('onUpdate', () => onUpdate(DragUpdateDetails(
           sourceTimeStamp: timestamp,
-          delta: _getDeltaForDetails(delta),
-          primaryDelta: _getPrimaryValueFromOffset(delta),
-          globalPosition: _initialPosition + delta,
+          delta: updateDelta,
+          primaryDelta: _getPrimaryValueFromOffset(updateDelta),
+          globalPosition: startPosition + delta,
```

Comment:
Won't this cause delta to be added twice in some cases? Specifically in the DragStartBehavior.start case, since we add delta to initialPosition to get startPosition?

Reply:
It seems like it might not make sense to send an onUpdate call from this function (acceptGesture), if the dragStartBehaviour is set to `start` because the delta is going into the onStart call.

Reply:
This code is guarded by the `updateDelta != Offset.zero` check, which won't pass for `DragStartBehavior.start` due to the switch logic above.

Result being that `onUpdate` is never called with `start` behavior; and when it is called (in `down` behavior), `startPosition` does not contain the delta, so there's no double counting.

I agree this flow is not the most direct and readable but it seemed best to me as it kept the code simpler. Happy to change if you think there's a better way.